src/Agent.Listener/Configuration/AutoLogonConfigurationManager.cs

```diff
@@ -0,0 +1,283 @@
+        {
+            AssertAdminAccess(true);
+            
+            /* We need to find out first if the AutoLogon was configured for the same user
+            If it is different user we should be reverting the AutoLogon user specific registries
+            */
+            var regManager = HostContext.GetService<IWindowsRegistryManager>();
+            var regHelper = new AutoLogonRegistryManager(regManager);
+            regHelper.FetchAutoLogonUserDetails(out string userName, out string domainName);
+            if(string.IsNullOrEmpty(userName) || string.IsNullOrEmpty(domainName))
+            {
+                throw new InvalidOperationException("AutoLogon is not configured on the machine.");
+            }
+
+            var nativeWindowsHelper = HostContext.GetService<INativeWindowsServiceHelper>();
+            if(nativeWindowsHelper.HasActiveSession(domainName, userName))
+            {
+                Trace.Info($"AutoLogon is enabled for the same user, reverting the registry settings now.");
+                regHelper.RevertOriginalRegistrySettings();
+            }
+            else
+            {
+                Trace.Info($"AutoLogon is enabled for the different user {domainName}\\{userName}, reverting the registry settings now.");
+                var securityIdForTheUser = nativeWindowsHelper.GetSecurityId(domainName, userName);
+                AutoLogonRegistryManager regHelperForDiffUser = new AutoLogonRegistryManager(regManager, securityIdForTheUser);
+                regHelperForDiffUser.RevertOriginalRegistrySettings();
+            }            
+        }
+
+        public bool IsAutoLogonConfigured()
+        {
+            //ToDo: Different user scenario
+            
+            //find out the path for startup process if it is same as current agent location, yes it was configured
+            var regHelper = new AutoLogonRegistryManager(HostContext.GetService<IWindowsRegistryManager>(), null);
+            var startupCommand = regHelper.GetStartupProcessCommand();
+
+            if(string.IsNullOrEmpty(startupCommand))
+            {
+                return false;
+            }
+
+            var expectedStartupProcessDir = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin));
+            return Path.GetDirectoryName(startupCommand).Equals(expectedStartupProcessDir, StringComparison.CurrentCultureIgnoreCase);
+        }
+
+        private void UpdateRegistriesForAutoLogon(AutoLogonRegistryManager regHelper, string userName, string domainName, string logonPassword)
+        {
+            regHelper.UpdateStandardRegistrySettings();
+
+            //auto logon
+            ConfigureAutoLogon(regHelper, userName, domainName, logonPassword);
+
+            //startup process
+            var startupProcessPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin), "agent.listener.exe");
+            var startupCommand = string.Format($@"{startupProcessPath} run");
```

Comment:
run.cmd should be the entry. I would do:
Process: Expand %ComSpec% and IOUtil.AssertFile
Args: `/D /S /C ""{pathToRunCmd}" run /runningAsAutoLogon"`